Bug fix for dropping multiple files onto canvas

shaoshing opened the pull request:
Fix for issue #3750 

We are receiving one `placeEvent` with information of the last new layer when dropping multiple files , which causes the layers panel to add only one new layer. This PR fix the issue by fetching all layer IDs of the current document and then compare with the previous document model to detect all new layer IDs.

iwehrman commented on src/js/actions/documents.js:
```diff
@@ -935,35 +934,70 @@ define(function (require, exports) {
+                            return this.transfer(layerActions.resetLayers, document, layer);
+                        })
+                        .then(function () {
+                            return this.transfer(layerActions.resetIndex, document);
```
I don't think we should ever need to call this more than once, but it seems from the control structure that it could actually be called N times for N placed layers, resulting in N^2 layers eventually reset. Should this be factored out and only called once? Or do we know for some reason that this condition can only hold for, say, one layer?

shaoshing commented:
Hi @iwehrman I've addressed your comment by make it more clear when resetting layers for an existed layer, which happens when user expands a LLSO that contains one layer and the layer is an embedded SO (more details about this scenario https://github.com/adobe-photoshop/spaces-design/issues/3145#issuecomment-151627443). Back to you for another review.

iwehrman commented:
Looks better, and the fix works for me!
